tensorboard/webapp/plugins/npmi/views/data_selection/metric_arithmetic/metric_arithmetic_element/metric_arithmetic_element_test.ts

```diff
@@ -0,0 +1,242 @@
+import {By} from '@angular/platform-browser';
+import {FormsModule, ReactiveFormsModule} from '@angular/forms';
+
+import {MatAutocompleteModule} from '@angular/material/autocomplete';
+import {MatInputModule} from '@angular/material/input';
+import {MatChipsModule} from '@angular/material/chips';
+
+import {Action, Store} from '@ngrx/store';
+import {provideMockStore, MockStore} from '@ngrx/store/testing';
+
+import {State} from '../../../../../../app_state';
+import {getMetricFilters} from '../../../../store/npmi_selectors';
+import * as npmiActions from '../../../../actions';
+import {appStateFromNpmiState, createNpmiState} from '../../../../testing';
+import {MetricArithmeticElementContainer} from './metric_arithmetic_element_container';
+import {MetricArithmeticElementComponent} from './metric_arithmetic_element_component';
+
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+
+describe('Npmi Metric Arithmetic Element Container', () => {
+  let store: MockStore<State>;
+  let dispatchedActions: Action[];
+  const css = {
+    FILTER_CHIP: '.filter-chip',
+    ELEMENT_REMOVE: '.mat-chip-remove',
+    INPUT: 'input',
+    VALUE_INVALID: '.value-invalid',
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [
+        MetricArithmeticElementContainer,
+        MetricArithmeticElementComponent,
+      ],
+      imports: [
+        FormsModule,
+        ReactiveFormsModule,
+        MatInputModule,
+        MatAutocompleteModule,
+        MatChipsModule,
+      ],
+      providers: [
+        provideMockStore({
+          initialState: appStateFromNpmiState(createNpmiState()),
+        }),
+      ],
+    }).compileComponents();
+
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+
+    dispatchedActions = [];
+    spyOn(store, 'dispatch').and.callFake((action: Action) => {
+      dispatchedActions.push(action);
+    });
+    store.overrideSelector(getMetricFilters, {
+      'npmi@test': {max: 1.0, min: -1.0, includeNaN: false},
+    });
+  });
+
+  it('renders npmi metric arithmetic element component', () => {
+    const fixture = TestBed.createComponent(MetricArithmeticElementContainer);
+    fixture.componentInstance.metric = 'npmi@test';
+    fixture.detectChanges();
+
+    const filterDiv = fixture.debugElement.query(By.css(css.FILTER_CHIP));
+    expect(filterDiv).toBeTruthy();
+  });
+
+  it('removes metric on click on remove button', () => {
+    const fixture = TestBed.createComponent(MetricArithmeticElementContainer);
+    fixture.componentInstance.metric = 'npmi@test';
+    fixture.detectChanges();
+
+    const removeButton = fixture.debugElement.query(By.css(css.ELEMENT_REMOVE));
+    removeButton.nativeElement.click();
+    expect(dispatchedActions).toEqual([
+      npmiActions.npmiRemoveMetricFilter({metric: 'npmi@test'}),
+    ]);
+
+    const inputs = fixture.debugElement.queryAll(By.css(css.INPUT));
+    expect(inputs.length).toBe(2);
```

Comment:
hmm, this assertion is not so obvious to me. Can you describe the intended behavior in the spec name?

Reply:
Ok, I dont understand why this is there myself. Removed it.